packages/next/src/server/app-render/dynamic-rendering.ts

```diff
@@ -662,6 +666,13 @@ export function throwIfDisallowedEmptyShell(
     )
   }
 
+  // If we've disabled throwing on empty static shell, then we don't need to
+  // track any dynamic access that occurs above the suspense boundary because
+  // we'll do so in the route shell.
+  if (dynamicValidation.doNotThrowOnEmptyStaticShell) {
+    return
+  }
+
```

Comment:
while this bypass looks similar to the one below this one is not intrinsic to the rules of dynamic. I think we should just lift this out and put it where we are choosing to `throwIfDisallowedEmptyStaticShell`.

It also means we don't need to track this boolean on the validation state and we don't need the boolean argument in the factory for this object.